yt_dlp/utils/_utils.py

```diff
@@ -1172,8 +1173,7 @@ def parse_iso8601(date_str, delimiter='T', timezone=None):
 
     date_str = re.sub(r'\.[0-9]+', '', date_str)
 
-    if timezone is None:
-        timezone, date_str = extract_timezone(date_str)
+    timezone, date_str = extract_timezone(date_str, timezone)
```

Comment:
> NOTE: I'm not sure why the default timezone overrides any that is extracted. No other extractors make use of this parameter. It seems to have been added years ago for an extractor that no longer uses it, and even then there was no need for it to override.

Reply:
The `timezone` parameter may not have been intended as a default, but as an override.

Reply:
It appears to be an "override" as that was probably the easiest method at the time for supplying a timezone for a timezone-less ISO8601 time string. The override behaviour was probably undefined/a bug, and changing it should not pose any compat issues.

It doesn't make much sense for it to be an override. If there is a timezone available in the date str, then it should use that timezone.  If you override it then the timestamp is wrong

https://github.com/yt-dlp/yt-dlp/commit/08b38d54015706d90d05371455ba86b7887cabcc